### ISSUE TITLE
Drop HHVM support

### ISSUE THREAD

**Issue by Hywan** [hoaproject/Central#69]
After a long discussion about https://github.com/hoaproject/Central/issues/36:

1. atoum has dropped support for HHVM,
2. No one has asked for HHVM support,
3. HHVM does not support PHP 7.

## Progression

  * [x] `Acl`,
  * [ ] `Compiler`,
  * [x] `Consistency`,
  * [x] `Console`,
  * [x] `Database`,
  * [x] `Dispatcher`,
  * [x] `Event`,
  * [x] `Exception`,
  * [x] `Graph`,
  * [x] `Iterator`,
  * [x] `Json`,
  * [x] `Locale`,
  * [x] `Mail`,
  * [ ] `Math`,
  * [x] `Mime`,
  * [ ] `Protocol`,
  * [x] `Registry`,
  * [x] `Ruler`,
  * [x] `Socket`,
  * [x] `Stream`,
  * [ ] `Ustring`,
  * [x] `View`,
  * [x] `Visitor`,
  * [x] `Websocket`,
  * [x] `Zombie`.

**Comment by Grummfy** [comment]
So we remove it totally from travis build?

**Comment by vonglasow** [comment]
@Grummfy Yes, it was decided during the last virtual meeting.

**Comment by Grummfy** [comment]
it was just be sure

**Comment by fredemmott** [comment]
3. What are you missing now? I'm not going to claim that it's enough for you to reconsider, but 3.20.2 has substantially improved support, including composer and phpunit in php7 mode.

**Comment by Hywan** [comment]
@fredemmott That's a good question. Most of the tools we are using to ensure the quality inside Hoa does not support HHVM, especially atoum (or PHP CS Fixer for “minor” tools —minor for us, the tool is not minor). There are PR on atoum to add more support for HHVM (https://github.com/atoum/atoum/issues/624, https://github.com/atoum/atoum/pull/625) but this is not the end of the journey. Most projects are dropping their support for HHVM, like Symfony and Doctrine. Doctrine is going to use Hoa, and some Symfony bundles are using Hoa too. We were going to support HHVM because of these projects, but since they have dropped HHVM, we no longer have any commitments. This is the same story with Laravel and PsySH —which uses Hoa— etc.

tl;dr: Hoa is used by a lot of projects, that find their ways into major frameworks, like Symfony, Laravel, Zend, Doctrine etc. All these projects have dropped HHVM support, so this is no longer a requirement for us.

**Comment by Hywan** [comment]
@fredemmott To be clear, this is not a move *against* HHVM. I am the lead dev of [Tagua VM](http://tagua.io/), an experimental PHP virtual machine. I know the amount of work it represents, and I can easily understand how it hurts when someone drops the support for your project. This is not *against* HHVM. This is us, with our limited resources, trying to provide the best we can to a large and attentive audience.

**Comment by fredemmott** [comment]
Yep, I'm just trying to get an idea of what PHP7 incompatibilities are most important, given the recent improvements and "HHVM does not support PHP 7."

**Comment by bobthecow** [comment]
> This is the same story with Laravel and PsySH —which uses Hoa— etc.

PsySH does support HHVM, for what it's worth. It's an "allowed failure" in our CI right now, because testing it on Travis has been a bit flakier than all the other PHP versions, but it's definitely supported:

https://travis-ci.org/bobthecow/psysh/jobs/252780783

**Comment by Hywan** [comment]
@bobthecow My mistake! Thank you for the notice!

**Comment by Grummfy** [comment]
for information the only not migrated library are the one that have no travis inside so :
* Compiler => https://github.com/hoaproject/Compiler/pull/67
* Math => https://github.com/hoaproject/Math/pull/56
* Protocol => https://github.com/hoaproject/Protocol/pull/20
* Ustring => https://github.com/hoaproject/Ustring/pull/40

**Comment by Hywan** [comment]
Great, thanks for your work everyone!
